node/test/go/dev-mode/test/integration/index.test.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const { expect } = require('chai');
+
+const path = require('path');
+const log = require('log').get('test');
+const cleanup = require('../lib/cleanup');
+const createCoreResources = require('../lib/create-core-resources');
+const processFunction = require('../lib/process-function');
+const resolveTestVariantsConfig = require('../lib/resolve-test-variants-config');
+
+for (const name of ['TEST_EXTERNAL_LAYER_FILENAME']) {
+  // In tests, current working directory is mocked,
+  // so if relative path is provided in env var it won't be resolved properly
+  // with this patch we resolve it before cwd mocking
+  if (process.env[name]) process.env[name] = path.resolve(process.env[name]);
+}
+
+describe('integration', function () {
+  this.timeout(120000);
+  const coreConfig = {};
+
+  const useCasesConfig = new Map([
+    [
+      '4sLogger',
+      {
+        variants: new Map([
+          ['v14', { configuration: { Runtime: 'nodejs14.x', Timeout: 5 } }],
+          ['v16', { configuration: { Runtime: 'nodejs16.x', Timeout: 5 } }],
+        ]),
+      },
+    ],
+  ]);
+
+  const testVariantsConfig = resolveTestVariantsConfig(useCasesConfig);
+
+  before(async () => {
+    await createCoreResources(coreConfig);
+    for (const testConfig of testVariantsConfig) {
+      testConfig.deferredResult = processFunction(testConfig, coreConfig).catch((error) => ({
+        // As we process result promises sequentially step by step in next turn, allowing them to
+        // reject will generate unhandled rejection.
+        // Therefore this scenario is converted to successuful { error } resolution
+        error,
+      }));
+    }
+  });
+
+  for (const testConfig of testVariantsConfig) {
+    it(testConfig.name, async () => {
+      const testResult = await testConfig.deferredResult;
+      if (testResult.error) throw testResult.error;
+      log.debug('%s test result: %o', testConfig.name, testResult);
+      const { expectedOutcome } = testConfig;
+      const { invocationsData } = testResult;
+      if (expectedOutcome === 'success' || expectedOutcome === 'error:handled') {
+        if (expectedOutcome === 'success' && !testConfig.isAsyncInvocation) {
+          for (const { responsePayload } of invocationsData) {
+            expect(responsePayload.raw).to.equal('"ok"');
+          }
+        }
+        for (const [, logs] of invocationsData.map((data) => data.logs).entries()) {
+          const logPayload = logs[0];
+          expect(logs.length).to.equal(8);
+          expect(logPayload.name).to.equal(testConfig.configuration.FunctionName);
+          logs.forEach((logItem, index) => {
+            const message = (logItem.messages || [{ message: '' }])[0].message;
+            expect(`${testConfig.name.split('-')[0]} ${index + 1}`).to.have.string(
+              message.slice(message.lastIndexOf('\t') + 1).replace('\n', '')
+            );
+          });
```

Comment:
This test looks specific to `4s-logger` fixture. I would configure it specifically in its configuration (it's likely we will add another test cases here in future, and then those tests as configured now will stand in a way).

You can do it by adding `config.test` aside `variants` as here:  https://github.com/serverless/console/blob/d2f87d796f99ee35f38769a26fe8e0f5d19dc456/node/packages/aws-lambda-sdk/test/integration/index.test.js#L993-L1000

While in here, where we confirm on some general outcome from any test, I'd probably just stick to confirming that function resolved with expected values (which can be same for all). e.g as it's done here: https://github.com/serverless/console/blob/d2f87d796f99ee35f38769a26fe8e0f5d19dc456/node/packages/aws-lambda-sdk/test/integration/index.test.js#L1084-L1087

Reply:
Ahh okay yeah this makes sense. I kept the "ok" test in the main body and moved the 4s-logger test up to the config map 👍